avocado-setup.py

```diff
@@ -199,23 +199,24 @@ def need_bootstrap(enable_kvm=False):
 def get_repo(repo, basepath):
     """
     To get given repo cloned/updated and install
-    :param repo: repo link
+    :param repo: tuple of repo link and branch(optional)
     :param basepath: base path where the repository has to be downloaded
     """
-    repo_name = repo.split('/')[-1].split('.')[0]
+    if not repo[1]:
+        branch = "master"
+    else:
+        branch = repo[1]
+    cmd_update = "b=%s;git reset --hard && git checkout master && git remote update && (git branch|grep $b||(git checkout $b && git switch -c $b))" % branch
```

Comment:
why are we using a `-c` here?

Reply:
when checkout to tag, it does not create a branch, rather detached from HEAD, so -c will help to create the new branch with the tag, and it works for branches aswell.

Reply:
```
[harish@localhost ~]$ git clone https://github.com/avocado-framework/avocado.git
Cloning into 'avocado'...
remote: Enumerating objects: 100, done.
...
Resolving deltas: 100% (31047/31047), done.
[harish@localhost ~]$ cd avocado
[harish@localhost avocado]$ git branch 
* master
[harish@localhost avocado]$ b='new-branch';git reset --hard && git checkout master && git remote update && (git branch|grep $b||(git checkout $b && git switch -c $b))
HEAD is now at ace4c915 Merge remote-tracking branch 'clebergnu/nrunner_std_size_limits'
Already on 'master'
Your branch is up to date with 'origin/master'.
Fetching origin
error: pathspec 'new-branch' did not match any file(s) known to git
[harish@localhost avocado]$ git branch 
* master
```
Seems branch has not been created. Anything I am missing here?